### ISSUE TITLE
Many unregistered IDSPACEs are in use

### ISSUE THREAD

**Issue by jamesaoverton** [OBOFoundry/purl.obolibrary.org#97]
In the past 24 hours I've seen requests for PURLs in these IDSPACEs that are not in the [OBO registry](https://github.com/OBOFoundry/OBOFoundry.github.io/tree/master/ontology). Some I recognize, but most I do not. (I compiled this list manually, so there may be some mistakes.)

- ABA
- ARG
- AcarDv
- B8467
- BILS
- CDT
- CGNC
- COLLECTION
- CS
- DERMO
- DHBA
- DrerDv
- EFO
- EKB
- ENSEMBL
- EXAMPLE
- EcoGene
- FB
- FBql
- FF
- FlyBase
- GENO
- GOCHE
- GgalDv
- HGNC
- HINO
- ID
- INO
- MBA
- MCBCC
- MGI
- MOBO
- MdomDv
- NCBIGene
- OARCS
- OBIws
- ODNAE
- OHD
- OMIABIS
- OMIM
- Orphanet
- PANTHER
- PBA
- PDO
- PHI
- PRO
- Pfam
- PomBase
- REO
- RGD
- SGD
- SIO
- SMTH
- TAIR
- UBPROP
- UCHEBI
- UniProtKB
- VICO
- XEO
- ZFIN
- ZP

**Comment by alanruttenberg** [comment]
A known problem that we're finally in a position to fix. We should go through these trying to find their owners and ensure they are registered properly. For ones that we can't identify we should write to obo-discuss with the list and warning that unless registered we will start to not let them through in some reasonable amount of time.

Of the ones there, I know
PRO - Not sure what to do - probably should find out who is using this and inform them to change to "PR_"
Pfam - I wonder if PRO is using these
OHD - Mine - thought I registered but I guess not. I will register it. Oral Health and Disease Ontology
CDT - Mine used in OHD. I will register.
OMIABIS - Contact Mathias Brochhausen mbrochhausen@users.sf.net
OBIws - Contact Jie Zheng
REO - Contact Melissa Haendel
SIO - Contact Michel Dumontier

There are others that are familiar but I'm not sure their status. Would it be possible to get a sampling of the domains that are requesting them, to see which are used generally and which might be used by only a small number of hosts?

I suspect a number of them are autogenerated from dbxrefs. 

Presumably these are falling through to berkeleybop?

Personally I'd like to get away from any automatic fallthrough like the one we are using now.

**Comment by jamesaoverton** [comment]
Yes, these are failing over to the old system. I'd like to eliminate the failover, but it looks like there's a lot to be done before that's feasible.

I'm logging the HTTP Referer for all requests (06521b1134be2f0380a803fd64cb2932ee93d110, see #63) but very few requests actually include a Referer. Ignoring #90, about half of the requests seem to be from Googlebot, and many more are from crawlers like "Baiduspider" and "Yahoo! Slurp".

**Comment by cmungall** [comment]
SIO has URIs like

semanticscience.org/resource/SIO_000230

If you're getting hits for $OBO/SIO_ then someone upstream of us has 
messed up the SIO URIs (hope it's not me).

**Comment by jamesaoverton** [comment]
@alanruttenberg I just noticed that UniProt uses PRO_ in at least some cases, for example the 'Feature identifier' column of the table here: http://www.uniprot.org/uniprot/P01308#ptm_processing

**Comment by cmungall** [comment]
These are not OBO-PR IDs!! This is UniProt's own system of identifiers 
that is uncoordinated with OBO-PR. This was AFAICR the basis for the 
original change from PRO->PR, to avoid confusion...

(you can tell the difference between a legacy PR and a UniProt-PRO by 
the number of digits)

**Comment by jamesaoverton** [comment]
I didn't know that. Thanks!

**Comment by jamesaoverton** [comment]
This Google Spreadsheet lists paths that are not being handled by the new PURL system:

https://docs.google.com/spreadsheets/d/171m26KaTkxJx_3Kf4nMiNTisINshdeUf9GgxTmW0Rcg/edit?usp=sharing

**Comment by nataled** [comment]
1) Chris M. is correct that the PRO idspace used to be 'PRO' but changed to 'PR' to avoid clashing with UniProt's 'PRO_' (at the time PRO began, these identifiers didn't exist, and PRO idspace was colon-delimited). The digits for an PR is 9, but for UniProt's PRO_ is 10.

2) PRO uses identifiers--and defines terms within the PRO as needed--for the following model organism databases (or other reference) for genes:
        CGNC
        EcoGene
        FlyBase
        HGNC
        MGI
        PomBase
        RGD
        SGD
        TAIR
        ZFIN
        NCBIGene (used for organisms that lack a MOD)

The notion here was that we don't create a new idspace for these well-established and authoritative resources. Not included on the list of unregistered-but-defined-by-PRO is dictyBase and wormBase.

3) PRO *cites* identifiers for other databases in cross-references or evidences, but does not define terms for:
        OMIM
        Pfam
        UniProtKB
        PANTHER
           ...and many others that are not on this 'unregistered' list. It's unclear if the list given includes only defined terms...? If so, then some other ontology is doing so.

4) UniProtKB is supposedly registered, unless you refer to registering a PURL. I know we went through some whole process of defining the regex for the identifiers. Alan was involved with this. As we don't 'own' any of the indicated resources, it is unclear if we can register the idspaces like we did for UniProtKB (The PRO PI is a co-PI of the UniProt project).

5) The URL for all the resources referenced by PRO is given in the PRO obo file.

**Comment by mcourtot** [comment]
I asked about EFO - proper URIs are http://www.ebi.ac.uk/efo/EFO_0000001, shouldn't find any OBO based ones.

GOCHE, MGI, Panther, SGD, RGD, TAIR, UniProtKB: @cmungall, are those GO related?

**Comment by cmungall** [comment]
I don't understand the question

**Comment by mcourtot** [comment]
I'm wondering if there is a chance those are somehow ending with an OBO type PURL via the GO. Maybe there is some link somewhere that adds a default OBO base to some IDs?

Re EFO: there was one (obsolete) term in the EFO file with an OBO PURL, which has now been cleaned.

**Comment by nataled** [comment]
With the exception of GOCHE, they are there because PRO references them (see my reply above). Only recently did I realize that OORT (which we've used to convert OBO to OWL) ignores the information in the header (where we indicate how the "ID:" prefix should be replaced), and instead automatically makes everything into an OBO purl. In the latest release of PRO we fixed such cases post-conversion.

inb4 "Use ROBOT" ;)

**Comment by cmungall** [comment]
Is there an action here? Do we bounce these back to n2t or a similar service?

**Comment by nataled** [comment]
I'm unfamiliar with that service. What does it do? As far the unregistered prefixes used by PRO, these will no longer default to an obolibrary PURL. That was fixed as of my previous comment of Oct 25 2016. I realize that the cited resources don't have a commitment to maintain the IRIs, but we do check and update as needed.

**Comment by jamesaoverton** [comment]
We need to review the logs again. If anybody has time to do this, let me know and I'll send you the log files.
